File: ruby/lib/turn_right_command.rb
class TurnRightCommand
	def initialize(robot)
		@robot = robot
    end

    def execute
		@robot.direction = Compass.get_right_turn_direction @robot.direction
    end

    def undo
		@robot.direction = Compass.get_left_turn_direction @robot.direction
	end
end

File: csharp/src/CommandPattern/TurnRightCommand.cs
namespace CommandPattern
{
    public class TurnRightCommand : ICommand
    {
        private Robot _robot;

        public TurnRightCommand(Robot robot)
        {
            _robot = robot;
        }

        public void Execute()
        {
            _robot.Direction = Compass.GetRightTurnDirection(_robot.Direction);
        }

        public void Undo()
        {
            _robot.Direction = Compass.GetLeftTurnDirection(_robot.Direction);
        }
    }
}